Edits to storage topic and light touch on supported releases (#3634)
cloudnative-pg/cloudnative-pg: docs/src/supported_releases.md

Supported releases of CloudNativePG include releases that are in the active
maintenance window and are patched for security and bug fixes.

> Subsequent patch releases on a minor release don't contain changes that are not backward
> compatible.

Reviewer: ```suggestion
Subsequent patch releases on a minor release contain backward compatible changes only.
```

Author: ```suggestion
Subsequent patch releases on a minor release contain backward-compatible changes only.

```

We produce new builds of CloudNativePG for each commit.

Approximately every two months, we build a minor release and run through several
additional tests as well as release qualification. We release patch versions
for issues found in minor releases.

The various types of releases represent a different product quality level and
level of assistance from the CloudNativePG community.
For details on the support provided by the community, see
[What we mean by support](#what-we-mean-by-support).

> | Type              | Support level                                                                                                         | Quality and recommended Uue                                                                                    |

Reviewer: ```suggestion
| Type              | Support level                                                                                                         | Quality and recommended Use                                                                                    |
```
